src/network/socket/Socket.hh

```diff
@@ -0,0 +1,33 @@
+<?hh // strict
+
+namespace HH\Lib\Experimental\Network;
+
+use type Throwable;
+
+interface Socket {
+  /**
+   * Close the underlying socket.
+   * 
+   * @param Throwable $e Reason for close, will be set as previous error.
+   */
+  public function close(?Throwable $e = null): void;
```

Comment:
If I'm understanding correctly, you're saying that Socket::close might throw a new exception, and can optionally set $e as the preivous exception:
-  is this correct?
- if so, it feels weird to only support it for close() when pretty mucha nything here (or in IO) could reasonably throw; why not remove it for consistency?
- if there's a special reason for it to be supported for close, please expand documentation

Reply:
its been removed.
in the php concurrent-phps `async-ext` , `$e` is passed to close and an exception is thrown when there's an await for read or write after the stream has been closed 
example :
```
$foo->writeAsync($someData); // no `await`
$foo->close(); // <-- exception thrown if `write` didn't finish ( optional `$e` will be set as a previous exception ) 
```
but since we don't support this in `HH\Lib\Io` i think there's no reason to do it here.